cypress/e2e/browse.cy.js

```diff
@@ -0,0 +1,166 @@
+describe('Browsing', () => {
+  let wares
+  let loading
+  let error
+  let intercepts = [
+    {
+      alias: 'useFilteredWares',
+    },
+    {
+      alias: 'useFilteredWares - blank search',
+    },
+    {
+      alias: 'useFilteredWares - with results',
+      query: 'test',
+    },
+    {
+      alias: 'useFilteredWares - no results',
+      query: 'asdfghjk',
+      emptyFixture: 'services/no-wares.json',
+    },
+  ]
+  
+  beforeEach(() => {
+    // Intercept the responses from the endpoint to view all requests.
+    // Even though this is to the same endpoint, the call happens on each page twice, 
+    // once when the page loads with all the wares, and again after any search is performed.
+    // this makes it necessary to create an intercept for each time the call is made.
+    intercepts.forEach((intercept) => {
+      cy.customApiIntercept({
+        action: 'GET',
+        alias: intercept.alias,
+        requestURL: `/providers/${Cypress.env('NEXT_PUBLIC_PROVIDER_ID')}/wares.json?q=${intercept.query || ''}`,
+        data: wares,
+        defaultFixture: 'services/wares.json',
+        emptyFixture: intercept.emptyFixture || '',
+        loading,
+        error
+      })
+    })
+  })
+
+  describe('from the browse page', () => {
+    beforeEach(() => {
+      cy.visit('/browse')
+    })
+
+    context('browse page is loading', () => {
+      before(() => {
+        loading = true
+      })
+      it('should show loading components.', () => {
+        cy.get(".card-body[data-cy='item-loading']").should('be.visible').then(() => {
+          cy.log('Loading components display correctly.')
+        })
+      })
+    })
+
+    context('error while making a request to the api', () => {
+      before(() => {
+        loading = false
+        error = true
+      })
+      it('should show an error message.', () => {
+        cy.get("div[role='alert']").should('be.visible').then(() => {
+          cy.log('Successfully hits an error.')
+        })
+      })
+    })
+
+    // filtering by query is handled in the api and not the webstore
+    // since we are stubbing that response, our tests will not return actual filtered wares based on on test query.
+    // the purpose of these tests is just to show that the correct components appear in the UI in each case.
+    context('a search is completed successfully and', () => {
+      before(() => {
+        wares = true
+        error = false
+      })
+      it('has a blank query', () => {
```

Comment:
tldr -- can you DRY this code by removing each of the comments within the "it" blocks on this page?

in general, code comments should explain _why_ something is being done. not _what_ is being done. the variable names, etc should be clear enough to explain the what. 

in testing, the strings inside the "context" and "it" blocks should be written well enough to explain what's going on. even more so... each test you've written perfectly describes what's going on. the comments are redundant.

Reply:
updated!